### PR TITLE
s3_bucket: accept NetApp StorageGRID create_bucket response.

### LINKED ISSUE #67459: s3_bucket: add StorageGRID support
##### SUMMARY

Support NetApp StorageGRID

##### ISSUE TYPE
- Feature Idea

##### COMPONENT NAME

s3_bucket

##### ADDITIONAL INFORMATION

StorageGRID provides S3 REST API, but it's not fully-compatible with AWS API.
Current StorageGRID 11.3 does not have Requester Pays and tags feature, furthermore, the API returns XNotImplemented error instead of the standard NotImplemented error.
http://docs.netapp.com/sgws-113/topic/com.netapp.doc.sg-s3/GUID-6E737B19-441A-495F-A35C-9BDC8BA1A4A2.html
Here is the error message that s3_bucket module in ansible 2.9.5 outputs:

```
An exception occurred during task execution. To see the full traceback, use -vvv. The error was: botocore.exceptions.ClientError: An error occurred (XNotImplemented) when calling the GetBucketRequestPayment operation: The request you provided implies functionality that is not implemented.
fatal: [localhost]: FAILED! => {"boto3_version": "1.12.0", "botocore_version": "1.15.0", "changed": false, "error": {"code": "XNotImplemented", "message": "The request you provided implies functionality that is not implemented.", "resource": "/ansible-bucket-0217?requestPayment"}, "msg": "Failed to get bucket request payment: An error occurred (XNotImplemented) when calling the GetBucketRequestPayment operation: The request you provided implies functionality that is not implemented.", "resource_actions": ["objectstorage-s:CreateBucket", "objectstorage-s:GetBucketVersioning", "objectstorage-s:ListBuckets", "objectstorage-s:GetBucketRequestPayment", "objectstorage-s:HeadBucket"], "response_metadata": {"host_id": "12889128", "http_headers": {"cache-control": "no-cache", "connection": "keep-alive", "content-length": "267", "content-type": "application/xml", "date": "Mon, 17 Feb 2020 08:15:45 GMT", "x-amz-id-2": "12889128", "x-amz-request-id": "1581927345613682",  "http_status_code": 501, "request_id": "1581927345613682", "retry_attempts": 0}}

```

### PR DESCRIPTION
**Pull request by r0ckpine**
##### SUMMARY

Support StorageGRID S3 API
Fixes #67459 

##### ISSUE TYPE

- Feature Pull Request

##### COMPONENT NAME

s3_bucket

##### ADDITIONAL INFORMATION

```paste below

```

### CODE REVIEW

File: lib/ansible/modules/cloud/amazon/s3_bucket.py [hunk 1]
```diff
@@ -247,7 +247,7 @@ def create_or_update_bucket(s3_client, module, location):
     except BotoCoreError as exp:
         module.fail_json_aws(exp, msg="Failed to get bucket request payment")
     except ClientError as exp:
-        if exp.response['Error']['Code'] != 'NotImplemented' or requester_pays:
+        if exp.response['Error']['Code']  not in ('NotImplemented', 'XNotImplemented') or requester_pays:
```
Comment (tremble) [inline lib/ansible/modules/cloud/amazon/s3_bucket.py]:
```suggestion
        if exp.response['Error']['Code'] not in ('NotImplemented', 'XNotImplemented') or requester_pays:
```

**Review by tremble — CHANGES_REQUESTED** [review]
Other than the sanity/linting issue, this PR looks reasonable and it doesn't seem to break the existing s3_bucket integration tests.

NetApp docs are https://docs.netapp.com/sgws-110/index.jsp?topic=%2Fcom.netapp.doc.sg-s3%2FGUID-6E737B19-441A-495F-A35C-9BDC8BA1A4A2.html and seem to imply that XNotImplemented is what it will throw for unsupported functions.

**Comment by r0ckpine** [comment]
ready_for_review

**Comment by tremble** [comment]
LGTM

Sorry about the s3_bucket test flakes, I've been investigating them.  They seem to be related to our use of us-east-1 for our CI.  (50 runs in us-west-2 and I couldn't trigger it, test in us-east-1 and I get a 15+% failure rate)

**Comment by tremble** [comment]
bot_status

**Comment by jillr** [comment]
Thanks @r0ckpine, this looks great but could you please add a [changelog](https://docs.ansible.com/ansible/latest/community/development_process.html#changelogs-how-to) noting the bugfix?

**Comment by r0ckpine** [comment]
ready_for_review

**Review by jillr — APPROVED** [review]

**Pull request merged**